### ISSUE TITLE
Manually deleted resouce fails all pulumi action

### ISSUE THREAD

**Issue by StarKitty** [pulumi/pulumi#2859]
A manually deleted resource locked my whole stack 
`pulumi refresh `
output 
`Previewing refresh (dev):

     Type                        Name                       Plan        Info
     pulumi:pulumi:Stack         aws-vpc-dev                            1 error
     ├─ aws:ec2:Vpc              Pulumi VPC                             
     ├─ aws:ec2:Subnet           Pulumi Subnet Public a                 
     ├─ aws:ec2:Subnet           Pulumi Subnet Public b                 
     ├─ aws:ec2:Subnet           Pulumi Subnet Private a                
     ├─ aws:ec2:Subnet           Pulumi Subnet Public c                 
     ├─ aws:ec2:Subnet           Pulumi Subnet Private c                
     ├─ aws:ec2:Subnet           Pulumi Subnet Private b                
     ├─ aws:ec2:Subnet           Pulumi DB-Subnet b                     
     ├─ aws:ec2:Subnet           Pulumi DB-Subnet a                     
     ├─ aws:ec2:InternetGateway  Pulumi IGW                             
     ├─ aws:ec2:RouteTable       Pulumi Public Route Table              
 -   ├─ aws:ec2:NetworkAcl       Public ACL                 delete      
     ├─ aws:ec2:Subnet           Pulumi DB-Subnet c                     
 ~   ├─ aws:ec2:NetworkAclRule   Port 22 Access             refresh     1 error
 ~   └─ aws:ec2:NetworkAclRule   Port 443 Access            refresh     1 error
 
Diagnostics:
  aws:ec2:NetworkAclRule (Port 22 Access):
    error: Preview failed: refreshing urn:pulumi:dev::aws-vpc::aws:ec2/networkAclRule:NetworkAclRule::Port 22 Access: Error Finding Network Acl Rule 120: InvalidNetworkAclID.NotFound: The networkAcl ID 'acl-08895192cd34381f2' does not exist
    	status code: 400, request id: e5794f56-6a56-422f-bb9a-7f37f94b8d59
 
  aws:ec2:NetworkAclRule (Port 443 Access):
    error: Preview failed: refreshing urn:pulumi:dev::aws-vpc::aws:ec2/networkAclRule:NetworkAclRule::Port 443 Access: Error Finding Network Acl Rule 110: InvalidNetworkAclID.NotFound: The networkAcl ID 'acl-08895192cd34381f2' does not exist
    	status code: 400, request id: 2f667947-236f-4436-8932-9377852bfaff
 
  pulumi:pulumi:Stack (aws-vpc-dev):
    error: preview failed
`
with -d 
`Previewing refresh (dev):

     Type                        Name                       Plan        Info
     pulumi:pulumi:Stack         aws-vpc-dev                            1 error; 32 debugs
     ├─ aws:ec2:Vpc              Pulumi VPC                             
     ├─ aws:ec2:Subnet           Pulumi Subnet Public b                 
     ├─ aws:ec2:Subnet           Pulumi Subnet Public a                 
     ├─ aws:ec2:Subnet           Pulumi Subnet Private a                
     ├─ aws:ec2:Subnet           Pulumi Subnet Public c                 
     ├─ aws:ec2:Subnet           Pulumi Subnet Private c                
     ├─ aws:ec2:Subnet           Pulumi Subnet Private b                
     ├─ aws:ec2:Subnet           Pulumi DB-Subnet b                     
     ├─ aws:ec2:Subnet           Pulumi DB-Subnet a                     
     ├─ aws:ec2:InternetGateway  Pulumi IGW                             
     ├─ aws:ec2:RouteTable       Pulumi Public Route Table              
 -   ├─ aws:ec2:NetworkAcl       Public ACL                 delete      
     ├─ aws:ec2:Subnet           Pulumi DB-Subnet c                     
 ~   ├─ aws:ec2:NetworkAclRule   Port 443 Access            refresh     1 error
 ~   └─ aws:ec2:NetworkAclRule   Port 22 Access             refresh     1 error
 
Diagnostics:
  aws:ec2:NetworkAclRule (Port 443 Access):
    error: Preview failed: refreshing urn:pulumi:dev::aws-vpc::aws:ec2/networkAclRule:NetworkAclRule::Port 443 Access: Error Finding Network Acl Rule 110: InvalidNetworkAclID.NotFound: The networkAcl ID 'acl-08895192cd34381f2' does not exist
    	status code: 400, request id: da68bb93-f83f-45d0-a49f-412384a40dfa
 
  pulumi:pulumi:Stack (aws-vpc-dev):
    debug: Setting AWS metadata API timeout to 100ms
    debug: Ignoring AWS metadata API endpoint at default location as it doesn't return any instance-id
    debug: No assume_role block read from configuration
    debug: Building AWS auth structure
    debug: Setting AWS metadata API timeout to 100ms
    debug: Ignoring AWS metadata API endpoint at default location as it doesn't return any instance-id
    debug: AWS Auth provider used: "EnvProvider"
    debug: Trying to get account information via sts:GetCallerIdentity
    debug: Trying to get account information via sts:GetCallerIdentity
    debug: Describing Network Acl: acl-08895192cd34381f2
    debug: Describing Network Acl with the Filters {
    debug:   Filters: [{
    debug:       Name: "entry.rule-number",
    debug:       Values: ["120"]
    debug:     },{
    debug:       Name: "entry.egress",
    debug:       Values: ["false"]
    debug:     }],
    debug:   NetworkAclIds: ["acl-08895192cd34381f2"]
    debug: }
    debug: Describing Network Acl: acl-08895192cd34381f2
    debug: Describing Network Acl with the Filters {
    debug:   Filters: [{
    debug:       Name: "entry.rule-number",
    debug:       Values: ["110"]
    debug:     },{
    debug:       Name: "entry.egress",
    debug:       Values: ["false"]
    debug:     }],
    debug:   NetworkAclIds: ["acl-08895192cd34381f2"]
    debug: }
    debug: Network ACL (acl-08895192cd34381f2) not found
    error: preview failed
 
  aws:ec2:NetworkAclRule (Port 22 Access):
    error: Preview failed: refreshing urn:pulumi:dev::aws-vpc::aws:ec2/networkAclRule:NetworkAclRule::Port 22 Access: Error Finding Network Acl Rule 120: InvalidNetworkAclID.NotFound: The networkAcl ID 'acl-08895192cd34381f2' does not exist
    	status code: 400, request id: c1976deb-b45e-48e9-85b1-f20a7464609c
 
`
the problem seems to be that even though there is no network ACL pulumi tries to discover the ACL rule is there any work around to dereference this resource or make Pulumi understand that this resource was deleted ?

**Comment by StarKitty** [comment]
So after some research, I found the command `pulumi state delete urn ` not sure it was the best way out but would be nice if pulumi could deal with this deletions on `pulumi refresh`

**Comment by ellismg** [comment]
> So after some research, I found the command pulumi state delete urn not sure it was the best way out but would be nice if pulumi could deal with this deletions on pulumi refresh

Yes, that is a but surprising, I would have expected that the AWS provider to say during refresh this resource no longer exists and so it should be reported as a delete.
